pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -390,20 +391,32 @@ func (r *ReconcileClusterDisruption) getOSDFailureDomains(clusterInfo *cephclien
 	for _, deployment := range osdDeploymentList.Items {
 		labels := deployment.Spec.Template.ObjectMeta.GetLabels()
 		failureDomainName := labels[topologyLocationLabel]
+		osdHostName := labels[osdHostNameLabel]
 		if failureDomainName == "" {
-			return []string{}, []string{}, errors.Errorf("failed to get the topology location label %q in OSD deployment %q",
+			return nil, nil, errors.Errorf("failed to get the topology location label %q in OSD deployment %q",
 				topologyLocationLabel, deployment.Name)
 		}
 
+		// Assume node drain if osd deployment ReadyReplicas count is 0 and osd node is not schedulable
 		if deployment.Status.ReadyReplicas < 1 {
-			if !drainingFailureDomains.Has(failureDomainName) {
-				drainingFailureDomains.Insert(failureDomainName)
+			isNodeDrained, err := isNodeDrained(r.client, osdHostName)
+			if err != nil {
+				return nil, nil, errors.Wrap(err, "failed to check if node is drained")
 			}
-		} else {
-			if !allFailureDomains.Has(failureDomainName) {
-				allFailureDomains.Insert(failureDomainName)
+			if isNodeDrained {
+				logger.Infof("osd %q is down and drain detected in node %q", deployment.Name, osdHostName)
+				if !drainingFailureDomains.Has(failureDomainName) {
+					drainingFailureDomains.Insert(failureDomainName)
+				}
+				continue
 			}
+			logger.Infof("osd %q is down but no drain detected in node %q", deployment.Name, osdHostName)
```

Comment:
How often will this print to the log? Should it be debug level?

Reply:
We'll see this on every reconcile and only if OSD is down but node is not draining. Looks like a useful information to be kept as `info` rather than `debug`.